lib/Types.cpp

```diff
@@ -317,8 +317,10 @@ Type *clspv::InferType(Value *v, LLVMContext &context,
           dyn_cast<MDString>(access_qual_op)->getString();
 
       if (access_qual_str == "none") {
-        assert(type_name_str.endswith("*") && "Only expect pointer types here");
-        auto type_name = type_name_str.drop_back(1);
+        auto type_name = type_name_str;
```

Comment:
the first bit of InferType is:
```
if (!isa<PointerType>(v->getType()) {
  return v->getType();
}
```

So how is this getting hit?

Reply:
I think I found the real bug.

We were missing a break, which lead to overriding `ordinal`